deltachat-ios/Controller/Settings/SingleSettings/VideoChatViewController.swift

```diff
@@ -31,23 +60,74 @@ class VideoChatViewController: UITableViewController {
     }
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 1
+        return self.defaultOptions.count + 1
     }
 
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        // selection changed, save value of custom field to ui config
+        dcContext.setConfig("ui.custom_webrtc_instance", customInstanceCell.getText())
+        
+        var newInstance: String?
+        if indexPath.row < self.defaultOptions.count {
+            newInstance = self.defaultOptions[indexPath.row].url
+            self.view.endEditing(true)
+        } else {
+            newInstance = customInstanceCell.getText()
+        }
+        dcContext.setConfig("webrtc_instance", newInstance)
+        
+        updateSelected()
+        
         tableView.deselectRow(at: indexPath, animated: true)
     }
+    
+    // called when editing the value of the custom field or clicking on it
+    @objc private func setCustom(_ textField: UITextField) {
+        let newInstance = customInstanceCell.getText()
+        dcContext.setConfig("ui.custom_webrtc_instance", newInstance)
+        dcContext.setConfig("webrtc_instance", newInstance)
+        updateSelected()
+    }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        videoInstanceCell.textField.text = dcContext.getConfig("webrtc_instance")
-        return videoInstanceCell
+        if indexPath.row < self.defaultOptions.count {
+            return self.defaultOptions[indexPath.row]
+        } else {
+            return customInstanceCell
+        }
+    }
+   
+    private var isCustom = true
+    
+    func updateSelected() {
+        let currentUrl = dcContext.getConfig("webrtc_instance")
+        // set selection
+        for option in self.defaultOptions {
+            if option.url == currentUrl {
+                option.accessoryType = .checkmark
+                self.isCustom = false
+            } else {
+                option.accessoryType = .none
+            }
+        }
+        
+        if self.isCustom {
+            customInstanceCell.accessoryType = .checkmark
+            customInstanceCell.textField.text = currentUrl
+        } else {
+            customInstanceCell.accessoryType = .none
+            customInstanceCell.textField.text = dcContext.getConfig("ui.custom_webrtc_instance")
+        }
     }
 
     override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
         return String.localized("videochat_instance_explain_2") + "\n\n" + String.localized("videochat_instance_example")
```

Comment:
wondering if the example string is still needed - `basicwebrtc`, as mentioned in the example string is not really a thing currently - and the jitsi example is shown already above.

Reply:
probably not, but I'm too lazy to remove it. maybe we should explain what `$ROOM` does instead?

Reply:
yip, let's keep the example string and update the string for all platforms at some point.